hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/messagevalidation/V2PayloadValidator.java

```diff
@@ -1,45 +1,179 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
 import org.apache.camel.Exchange;
 import org.apache.camel.Handler;
+
+import org.eclipse.jetty.util.StringUtil;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.Set;
+import ca.bc.gov.hlth.hnsecure.authorization.AuthorizationProperties;
+import ca.bc.gov.hlth.hnsecure.message.ErrorMessage;
+import ca.bc.gov.hlth.hnsecure.message.ErrorResponse;
+import ca.bc.gov.hlth.hnsecure.message.HL7Message;
+import ca.bc.gov.hlth.hnsecure.message.MessageUtil;
+import ca.bc.gov.hlth.hnsecure.parsing.Util;
+import net.minidev.json.JSONObject;
+import net.minidev.json.parser.JSONParser;
 
 public class V2PayloadValidator {
 
-    private static final Logger logger = LoggerFactory.getLogger(V2PayloadValidator.class);
+	private static final Logger logger = LoggerFactory.getLogger(V2PayloadValidator.class);
+
+	private static Set<String> validV2MessageTypes;
+	private static Set<String> validReceivingFascility;
+	private static String processingDomain;
+	private static final String expectedEncodingChar = "^~\\&";
+	private static boolean isValidSeg;
+	private static final String segmentIdentifier = "MSH";
+	private static final List<String> validDomainType = List.of(new String[]{"P", "E", "T", "D"});
+	private static ErrorMessage errorMessage;
+	private static ErrorResponse errorResponse;
+
+	private static final JSONParser jsonParser = new JSONParser(JSONParser.DEFAULT_PERMISSIVE_MODE);
+
+	public V2PayloadValidator(AuthorizationProperties authorizationProperties) {
+		validV2MessageTypes = authorizationProperties.getValidV2MessageTypes();
+		validReceivingFascility = authorizationProperties.getValidReceivingFacility();
+		processingDomain = authorizationProperties.getProcessingDomain();
+	}
+
+	/**
+	 * Validates the Hl7V2 transaction type (MSH.8) format and required fields
+	 * 
+	 * @param v2Message the hl7v2 message to validate
+	 */
+	@Handler
+	public static void validate(Exchange exchange, String v2Message) {
+
+		isValidSeg = true;
+		HL7Message messageObj = new HL7Message();
+		errorResponse = new ErrorResponse();
+
+		String auth = (String) exchange.getIn().getHeader("Authorization");
+
+		// Validate v2Message format
+		if (!StringUtil.isEmpty(v2Message)) {
+			String[] v2DataLines = v2Message.split("\r\n");
+			String[] v2Segments = v2DataLines[0].split("\\|");
+
+			if (Arrays.stream(v2Segments).allMatch(Objects::nonNull) && v2Segments.length >=12) {
+				errorResponse.initSegment(v2Segments, messageObj);
+			} else {
+				isValidSeg = false;
+				errorMessage = ErrorMessage.HL7Error_Msg_InvalidHL7Format;
+
+			}
+		} else {
+			isValidSeg = false;
+			errorMessage = ErrorMessage.HL7Error_Msg_NoInputHL7;
+
+		}
+
+		// Validate segment identifier
+		if (isValidSeg && !messageObj.getSegmentIdentifier().equals(segmentIdentifier)) {
+			isValidSeg = false;
```

Comment:
I think when a validation fails we want to just update the exchange body with the new response and throw an exception that can be handled in the route, something like: 

```
if (some validation fails) {
    exchange.setBody(createErrorResponse(errorMessage.SOME_ERROR_MSG));
    throw new ValidationFailedException();
}
```